### ISSUE TITLE
Deep clone of state in reducers?

### ISSUE THREAD

**Issue by dradovic** [mrpmorris/blazor-fluxor#61]
This is just a question. Does blazor-fluxor offer something for deep copying states as required in the reducers? Or can you recommend another library/pattern for this? Basically, I'm looking for something like [Immer](https://github.com/immerjs/immer).

Thank you very much for this wonderful library.

**Comment by andrevlins** [comment]
I'm using https://github.com/force-net/DeepCloner for this purpose.

Hope it helps you

**Comment by asalvo** [comment]
I'm using https://github.com/AlenToma/FastDeepCloner. I picked it because it had recent development and a fair number of downloads when searching for a deep cloning package on NuGet. It worked pretty much out of the box (just needed to pass FieldType.Both to the Clone extension method) with my State objects which have private fields (for exposing ReadOnly collections) and protected setters on properties. 

I did not try any other libraries as once I saw it worked for my State objects that was good enough for me at this point for my project. 

I just looked at DeepCloner and he published benchmarks showing it to be much faster then FastDeepCloner, but the results may be outdated (this [SO](https://stackoverflow.com/questions/852064/faster-deep-cloning) post he has FastDeepCloner benchmarked at 1800ms which is still slower then DeepCloner but much better then 15000ms). If I feel that performance starts to become an issue I'll re-evaluate.
